src/main/webapp/resources/js/pages/projects/samples/linker/Linker.jsx

```diff
@@ -27,58 +29,85 @@ const CommandText = styled(Paragraph)`
  * @constructor
  */
 function Linker() {
-  function handleSampleIds(e) {
-    // Post data to get linker command.
-    const { detail } = e;
+  const [visible, setVisible] = useState(false);
+  const [types, setTypes] = useState(["fastq"]);
+  const [scriptString, setScriptString] = useState();
+  const [command, setCommand] = useState();
+
+  const options = [
+    { label: i18n("Linker.fastq"), value: "fastq" },
+    { label: i18n("Linker.assembly"), value: "assembly" }
+  ];
+
+  function handleSampleIds({ detail }) {
     getNGSLinkerCode(detail).then(({ data }) => {
-      Modal.success({
-        className: "t-linker-modal",
-        width: 500,
-        title: i18n("Linker.title"),
-        content: (
-          <>
-            <Paragraph>{i18n("Linker.details")}</Paragraph>
-            <Text type="secondary">
-              <span dangerouslySetInnerHTML={{ __html: i18n("Linker.note") }} />
-            </Text>
-            <CommandText
-              className="t-cmd-text"
-              ellipsis={{ rows: 1 }}
-              copyable={{ text: data }}
-            >
-              {data}
-            </CommandText>
-          </>
-        )
-      });
+      // Post data to the server to get the linker command.
+      setScriptString(data);
+      setVisible(true);
     });
   }
 
+  useEffect(() => {
+    setCommand(
+      types.length ? `${scriptString} -t ${types.join(",")}` : scriptString
+    );
+  }, [scriptString, types]);
+
   /*
   These Listeners and Dispatchers are a way to get around the separation between react
   components and the legacy JS code already on the page.
   */
   document.addEventListener("sample-ids-return", handleSampleIds, false);
 
-  function showModal() {
+  function getIds() {
     document.dispatchEvent(new Event("sample-ids"));
   }
 
   return (
-    <Button
-      type="link"
-      style={{
-        margin: `0 inherit`,
-        padding: 0,
-        color: grey9
-      }}
-      className="t-linker-btn"
-      onClick={showModal}
-    >
-      {/* constant marginRight here to match old styles */}
-      <IconCode style={{ marginRight: 2 }} />
-      {i18n("project.samples.export.linker")}
-    </Button>
+    <>
+      <Button
+        type="link"
+        style={{
+          margin: `0 inherit`,
+          padding: 0,
+          color: grey9
+        }}
+        className="t-linker-btn"
+        onClick={getIds}
+      >
+        {/* constant marginRight here to match old styles */}
+        <CodeOutlined style={{ marginRight: 2 }} />
```

Comment:
Replace this with `IconCode` it's already in the imports, but I must have missed updating it.  Please then remove the import for `CodeOutlined`

Reply:
Done in 6cff5b9